Unity-Technologies/Graphics
# [Draft] Fix Flipbook Node

*Draft due to desire to find a good way to test without adding 600+ materials to the Test Project*

*Summary:*

- Fix for https://fogbugz.unity3d.com/f/cases/1219169/ (see automation for before & after!)
- Originally I had looked at a more complex solution involving dynamic solutions, I didn't go that way however I kept an optimization from it, which will counteract the performance hit the fix requires so that no one can complain about this fix (and overall performance win).
- Many materials for test scene: import time of these materials was less than 1 second on my machine.

*Manually Tested:*

Working5x5Flipbook0
Various node set ups, including:
- 5x5, 6x6, 5x7, 7x7, 7x5
- Using time as an input as real world use case would be
- Half & Float precision
- Negative values

Technical Risk: 3/4 - Only affects this node, but in a big way
Halo Effect: 0/4 - Only affects this node

*Notes to Documentation:*

The user no longer has to floor the Tile input before putting inserting it into the node. From the documentation, this kind of seems expected/implied anyways, so I didn't change anything. Thoughts?

*Notes to QA:*

- If you want to be more sure of this fix & have the time, I would use a real world use case on one of the grid sizes where the bug used to reproduces that isn't mentioned in my manual testing (ex: did it use to reproduce with 11x11? idk, but if it did you could try that. But that's all just an suggestion.)
